libbeat/processors/add_locale/add_locale.go

```diff
@@ -1,29 +1,99 @@
 package actions
 
 import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/processors"
 )
 
-type addLocale struct{}
+type addLocale struct {
+	TimezoneFormat TimezoneFormat
+}
+
+type TimezoneFormat int
+
+const (
+	Abbrevation TimezoneFormat = iota
+	Offset
+)
+
+var timezoneFormats = [...]string{
+	"abbrevation",
+	"offset",
+}
+
+func (t TimezoneFormat) String() string {
+	return timezoneFormats[t]
+}
 
 func init() {
 	processors.RegisterPlugin("add_locale", newAddLocale)
 }
 
 func newAddLocale(c common.Config) (processors.Processor, error) {
-	return addLocale{}, nil
+	config := struct {
+		Format string `config:"format" validate:"required"`
+	}{}
```

Comment:
I think it should default to offset. IMO that would be easier to use and not require someone to have a lookup table of names to numbers and account for DST changes.